test/e2e/templates_test.go

```diff
@@ -0,0 +1,138 @@
+	kubeConfig := e2e.NewClientConfigForTest(t)
+
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	routeClient, err := routeclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	configSecretsClient := kubeClient.CoreV1().Secrets("openshift-config")
+
+	cleanup := createSecret(t, configSecretsClient, "login", "login.html", "login test")
+	defer cleanup()
+	cleanup = createSecret(t, configSecretsClient, "providers", "providers.html", "provider selection test")
+	defer cleanup()
+	cleanup = createSecret(t, configSecretsClient, "error", "errors.html", "error test")
+	defer cleanup()
+	cleanup = createSecret(t, configSecretsClient, "htpasswd1", "htpasswd", "test:$2y$05$9Co/ojOvEs6IZUTxAdlHbO8leelkkmcwPMtlGTHFkxcTLrC86EbLG") // test:password
+	defer cleanup()
+
+	oauthConfig, err := configClient.ConfigV1().OAuths().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	oauthConfigCopy := oauthConfig.DeepCopy()
+
+	oauthConfigCopy.Spec.Templates.Login.Name = "login"
+	oauthConfigCopy.Spec.Templates.ProviderSelection.Name = "providers"
+	oauthConfigCopy.Spec.Templates.Error.Name = "error"
+
+	oauthConfigCopy.Spec.IdentityProviders = []configv1.IdentityProvider{
+		{
+			Name: "super duper production-ready htpasswd idp",
+			IdentityProviderConfig: configv1.IdentityProviderConfig{
+				Type: configv1.IdentityProviderTypeHTPasswd,
+				HTPasswd: &configv1.HTPasswdIdentityProvider{
+					FileData: configv1.SecretNameReference{
+						Name: "htpasswd1",
+					},
+				},
+			},
+		},
+	}
+
+	_, err = configClient.ConfigV1().OAuths().Update(context.TODO(), oauthConfigCopy, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	defer func() {
+		oauthConfigNew, err := configClient.ConfigV1().OAuths().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		require.NoError(t, err)
+
+		oauthConfigNew.Spec.IdentityProviders = oauthConfig.Spec.IdentityProviders
+		oauthConfigNew.Spec.Templates = oauthConfig.Spec.Templates
+
+		_, err = configClient.ConfigV1().OAuths().Update(context.TODO(), oauthConfigNew, metav1.UpdateOptions{})
+		require.NoError(t, err)
+	}()
+
+	// wait for new rollout
+	err = e2e.WaitForClusterOperatorProgressing(t, configClient.ConfigV1(), "authentication")
+	require.NoError(t, err, "authentication operator never became progressing")
+
+	err = e2e.WaitForClusterOperatorAvailableNotProgressingNotDegraded(t, configClient.ConfigV1(), "authentication")
+	require.NoError(t, err, "failed to wait for the authentication operator to become available")
+
+	route, err := routeClient.RouteV1().Routes("openshift-authentication").Get(context.TODO(), "oauth-openshift", metav1.GetOptions{})
+	require.NoError(t, err)
+	oauthURL, err := url.Parse("https://" + route.Spec.Host)
+	require.NoError(t, err)
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
```

Comment:
(no action required) Do we have any test environment in our e2e that needs this flag to be set? Does this need to be set for all test involving oauth-server?

Reply:
I'd rather have it set in case someone wants to try running the tests with a proxy. It definitely needs to appear in origin tests that involve http calls outside the cluster/to routes.

Reply:
Interesting, Good to know. Thanks!